File: app/src/main/java/net/daverix/ajvm/jvm/IndexedFStoreOperation.java
package net.daverix.ajvm.jvm;


import net.daverix.ajvm.Frame;
import net.daverix.ajvm.ByteCodeReader;

import java.io.IOException;

public class IndexedFStoreOperation implements ByteCodeOperation {
    private final int index;

    public IndexedFStoreOperation(int index) {
        this.index = index;
    }

    @Override
    public void execute(ByteCodeReader reader,
                        int indexOfBytecode,
                        Frame currentFrame) throws IOException {
        Object localVariable = currentFrame.pop();
        if(!(localVariable instanceof Float))
            throw new IllegalStateException("variable " + localVariable + " is not a Float");

        currentFrame.setLocalVariable(index, localVariable);
    }
}

File: app/src/main/java/net/daverix/ajvm/operation/IndexedFStoreOperation.kt
package net.daverix.ajvm.operation


import net.daverix.ajvm.Frame
import net.daverix.ajvm.ByteCodeReader

import java.io.IOException

class IndexedFStoreOperation(private val index: Int) : ByteCodeOperation {

    @Throws(IOException::class)
    override fun execute(reader: ByteCodeReader,
                         indexOfBytecode: Int,
                         currentFrame: Frame) {
        val localVariable = currentFrame.pop()
        if (localVariable !is Float)
            throw IllegalStateException("variable $localVariable is not a Float")

        currentFrame.setLocalVariable(index, localVariable)
    }
}
